File: parent-group4/pingyougou-web-shop/src/main/webapp/js/service/specCheckService.js
// 定义服务层:
app.service("specCheckService",function($http){
	this.findAll = function(){
		return $http.get("../specCheck/findAll.do");
	}
	
	this.findPage = function(page,rows){
		return $http.get("../specCheck/findPage.do?pageNum="+page+"&pageSize="+rows);
	}
	
	this.add = function(entity){
		return $http.post("../specCheck/add.do",entity);
	}

    this.update=function(entity){
        return $http.post("../specCheck/update.do",entity);
    }

    this.findOne=function(id){
        return $http.get("../specCheck/findOne.do?id="+id);
    }

    this.dele = function(ids){
        return $http.get("../specCheck/delete.do?ids="+ids);
    }
	
	// this.search = function(page,rows,searchEntity){
	// 	return $http.post("../brandCheck/search.do?pageNum="+page+"&pageSize="+rows,searchEntity);
	// }
	
	this.selectOptionList = function(){
		return $http.get("../specCheck/selectOptionList.do");
	}
});

File: parent-group4/pingyougou-interface/src/main/java/com/itheima/core/SpecCheckService.java
package com.itheima.core;

import com.itheima.core.pojo.specification.SpecificationCheck;
import entity.PageResult;
import pojogroup.SpecificationVo;

import java.util.List;
import java.util.Map;

public interface SpecCheckService {
    void add(SpecificationVo specificationVo, String name);

    SpecificationVo findOne(Long id);

    void update(SpecificationVo specificationVo);

    void delete(Long[] ids);

    PageResult search(Integer page, Integer rows, SpecificationCheck specificationCheck, String name);

    List<Map> selectOptionList(String name);

}
